atlasdb-workload-server-api/src/main/java/com/palantir/atlasdb/workload/workflow/WorkflowHistoryValidator.java

```diff
@@ -18,13 +18,17 @@
 
 import com.palantir.atlasdb.workload.invariant.InvariantReporter;
 import java.util.List;
+import org.immutables.value.Value;
 
-/**
- * Run the provided workflow and execute the provided invariant reporters.
- *
- * The idea of this class is that long-term we will want to have multiple ways to run/execute workflows,
- * rather than biasing towards the present case (just once).
- */
-public interface WorkflowRunner<WorkflowTypeT extends Workflow> {
-    void run(WorkflowTypeT workflow, List<InvariantReporter<?>> invariantReporters);
+@Value.Immutable
+public interface WorkflowHistoryValidator {
+    @Value.Parameter
+    WorkflowHistory history();
+
+    @Value.Parameter
+    List<InvariantReporter<?>> invariants();
+
+    static WorkflowHistoryValidator of(WorkflowHistory history, List<InvariantReporter<?>> invariants) {
```

Comment:
Considered making a static constructor which allowed `WorkflowValidator` to `WorkflowHistoryValidator`, but decided against it as naming was difficult, plus `Workflow::run` is a heavy method.